src/puffin/src/puffin_manager/stager/bounded_stager.rs

```diff
@@ -332,6 +395,9 @@ impl BoundedStager {
         }
 
         for (key, value) in elems {
+            if let Some(notifier) = self.notifier.as_ref() {
+                notifier.on_cache_insert(value.size());
+            }
             self.cache.insert(key, value).await;
         }
```

Comment:
```suggestion
        if let Some(notifier) = self.notifier.as_ref() {
            for (key, value) in elems {
                notifier.on_cache_insert(value.size());
                self.cache.insert(key, value).await;
            }
        } else {
            for (key, value) in elems {
                self.cache.insert(key, value).await;
            }
        }
```

Perhaps it's better to avoid checking the option in the for loop.